src/System/Taffybar/WorkspaceHUD.hs

```diff
@@ -337,10 +338,10 @@ buildWorkspaces _ = ask >>= \context -> liftX11Def M.empty $ do
   activeWindows <- readAsListOfWindow Nothing "_NET_ACTIVE_WINDOW"
   active:visible <- getVisibleWorkspaces
   let getWorkspaceState idx ws
-        | urgentWorkspaceState (hudConfig context) && not (null urgentWindows) =
-          Urgent
         | idx == active = Active
         | idx `elem` visible = Visible
+        | urgentWorkspaceState (hudConfig context) && not (null (ws `intersect` urgentWindows)) =
```

Comment:
so this makes it so that the "active" property of workspaces takes precedence over the urgent property. Now that I think about it, I guess that seems more correct.

I wonder if in the future we should make it possible to configure this, although I'm fine with it for now.

Reply:
My experience with the urgency setting before this change was that I'd be typing into vim, which likes to make a lot of beeps, and with the urxvt `urgentOnBell` option on, my active workspace to be marked urgent every few seconds.  That should probably be fixed by determining workspace urgency without considering active windows (something like `urgentWindows <- filterM isWindowUrgent (wins \\ activeWindows)`).  Then, I think it would be reasonable to want to be able to configure whether urgency takes precedence over active and/or visible states.  My opinion is that if you can see the window, you don't need a workspace notification about it, but I could understand how others would want a notification.  I will open an issue to discuss this further.